packages/block-editor/src/components/link-control/index.js

```diff
@@ -147,7 +148,7 @@ function LinkControl( {
 							} }
 							isSelected={ index === selectedSuggestion }
 							isURL={ manualLinkEntryTypes.includes( suggestion.type.toLowerCase() ) }
-							searchTerm={ inputValue }
+							searchTerm={ searchText }
```

Comment:
> E.g. "About" value should return same results as "About ".

If this is true, shouldn't we be intercepting the suggestions request value and trimming it instead so we avoid request URLs like this?

`rest_route=search=page%20%20%20`

We could do it here for example: https://github.com/WordPress/gutenberg/compare/master...ramonjd:add/url-input-trim-input-value#diff-7f80e90a821240f277aae4a7d2d9e778R172

The result:

![Jan-30-2020 13-28-08](https://user-images.githubusercontent.com/6458278/73506859-0e68a980-437b-11ea-8e1b-2e956f999836.gif)

We get the same results for `'about'`, `'    about'` and `'about    '`

Having said all this, if "About" value **_doesn't have to_** return same results as "About ", then ignore me :)

Reply:
To avoid that, the results are a bit different. 

* No space in input -> show URL item
* space in input -> doesn't show URL item because it's invalid URL.

Reply:
> If this quote from #19631 is true, shouldn't we be intercepting the suggestions request value and trimming it instead so we avoid request URLs like this?

> rest_route=search=page%20%20%20

@ramonjd, how can I reproduce this issue?

> We get the same results for 'about', ' about' and 'about '

IMO, this would be the correct behavior. One comment to the implementation - if the search term doesn't change (`"about"` === `"about   "`), we shouldn't fire another search query.

Reply:
> how can I reproduce this issue?

Hi @WunderBart 

I was just querying page names in the input field, and checking the network requests:

<img width="704" alt="Screen Shot 2020-02-05 at 10 54 35 am" src="https://user-images.githubusercontent.com/6458278/73798138-1787c980-4806-11ea-8b1a-7067848bf8cd.png">

Reply:
> One comment to the implementation - if the search term doesn't change ("about" === "about "), we shouldn't fire another search query. The trimmed value should be compared before the call.

I guess this could also be done in `onChange` in [UrlInput](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/url-input/index.js#L210)

Reply:
> I was just querying page names in the input field, and checking the network requests:

Ah, I didn't get it the first time - all clear now, thanks! That's exactly what I meant, too 😄

Reply:
> I guess this could also be done in onChange in UrlInput

I agree with this. See https://github.com/WordPress/gutenberg/pull/19832#discussion_r378782848